### PR TITLE
[arg-router] Update to v1.2.0

### PR DESCRIPTION
**Pull request by cmannett85**
## [arg_router v1.2.0](https://github.com/cmannett85/arg_router)

Bug fixes
* Issue 254, passing no args to short_form_expander_t::pre_parse_phase(..) causes segfault
* Issue 255, policy_unique_from_owner_parent_to_mode_or_root not working

Improvements
* Corrected CMake package so you don't need to manually set the include directory anymore (Issue 257)
* Node can be instantiated with bare compile-time strings and they are automatically mapped to the appropriate policy (Issue 256)
* Improved help output formatting documentation (Issue 258)

----------------------------------------------------------------------------------------------
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] SHA512s are updated for each updated download
- [x] The "supports" clause reflects platforms that may be fixed by this new version
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning ./vcpkg x-add-version --all and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by cmannett85** [comment]
You're right, I blindly copied the line from a real project that had `find_package(Boost)` call in it. `arg_router`'s CMake package is supposed to check the dependency itself but I'm still working on that - it doesn't matter in the context of vcpkg though as that dependency is managed via `vcpkg.json` so the required Boost headers are always there.

**Comment by cmannett85** [comment]
@MonicaLiu0311 I've updated the patch but I can't remove the `requires:author-response` label.

**Comment by MonicaLiu0311** [comment]
Using VS2022 to test the new `usage`, an error occurred during `build`, see the attachment for details.

[output.txt](https://github.com/microsoft/vcpkg/files/10846099/output.txt)

File: ports/arg-router/portfile.cmake [hunk 1]
```diff
@@ -21,17 +21,23 @@ file(COPY "${CMAKE_CURRENT_LIST_DIR}/usage"
...
+if (WIN32)
+    set(CMAKE_FILE_DIR "cmake")
+elseif (APPLE)
+    set(CMAKE_FILE_DIR "arg_router.framework/Resources/CMake")
+endif()
+file(RENAME "${CURRENT_PACKAGES_DIR}/${CMAKE_FILE_DIR}" "${CURRENT_PACKAGES_DIR}/share/arg_router")
+
 vcpkg_cmake_config_fixup(
     PACKAGE_NAME arg_router
 )
```
Comment (dg0yt) [inline ports/arg-router/portfile.cmake:34]: This command has a `CONFIG_PATH` option which is responsible for moving the config files and fixing paths.

**Comment by cmannett85** [comment]
> Using VS2022 to test the new `usage`, an error occurred during `build`, see the attachment for details.
> 
> [output.txt](https://github.com/microsoft/vcpkg/files/10846099/output.txt)

This library requires a minimum of C++17, MSVC defaults to C++14 so you'll need to add `target_compile_features(CMakeFindUsage PUBLIC cxx_std_17)` (or higher) to your `CMakeLists.txt`.  I don't think it's appropriate for me to add toolchain related settings to the usage file, the minimum C++ version is specified in the library documentation, a link to which is in the usage file.

File: versions/a-/arg-router.json [hunk 2]
```diff
@@ -11,4 +16,4 @@
       "port-version": 0
     }
   ]
-}
+}
```
Comment (dg0yt) [inline versions/a-/arg-router.json:19]: Unexpected change.

**Review by dg0yt — COMMENTED** [review]
Extra notes for upstream:
- It would have been better if the exported target would use a namespace (`arg_router::...`). Cmake could recognize it is a target, and clearly and early notify the user if he forgot `find_package()`.
- It would be good if some variables could be turned into cache variables (aka input variables), in particular install locations e.g. for cmake config.

**Comment by cmannett85** [comment]
> * It would have been better if the exported target would use a namespace (`arg_router::...`). Cmake could recognize it is a target, and clearly and early notify the user if he forgot `find_package()`.

I avoided using a namespace simply because there is only one exported target from the package, but you're right, it doesn't fit the pattern of other packages so I'll update it for the next release (https://github.com/cmannett85/arg_router/issues/275).

> * It would be good if some variables could be turned into cache variables (aka input variables), in particular install locations e.g. for cmake config.

In a 'normal' installation CMake would look in [certain platform-specific locations](https://cmake.org/cmake/help/latest/command/find_package.html#config-mode-search-procedure) for the package config files, so I never considered making this settable by the user.  But obviously `vcpkg` changes this, so I'll add one in (https://github.com/cmannett85/arg_router/issues/276).

Thanks for your feedback!

File: versions/a-/arg-router.json (hunk 2 above)
Comment (cmannett85) [reply versions/a-/arg-router.json]: My IDE being 'helpful' - I've replaced the newline.

**Comment by cmannett85** [comment]
> This is merely a convention, so all (W) and (U) directories are still searched on all platforms.

**Comment by MonicaLiu0311** [comment]
The usage test passed (header files found): 
```
The package arg-router is a header-only library and so is typically used like this:

    find_package(arg_router REQUIRED)
    include_directories(SYSTEM "${arg_router_INCLUDE_DIRS}")
```

**Comment by cmannett85** [comment]
@MonicaLiu0311 the `include_directories` call is not necessary, as the package provides an exported target which already has the include directory as a property.  So your test `CMakeLists.txt` should be:
```
cmake_minimum_required (VERSION 3.8)
set(CMAKE_TOOLCHAIN_FILE "E:/arg-router/scripts/buildsystems/vcpkg.cmake")

project ("CMakeFindUsage")
find_package(arg_router REQUIRED)

add_executable (CMakeFindUsage "CMakeFindUsage.cpp")
target_link_libraries(CMakeFindUsage PUBLIC arg_router)
target_compile_features(CMakeFindUsage PUBLIC cxx_std_17)
```

**Review by MonicaLiu0311 — APPROVED** [review]

**Pull request merged**